tools/offline_ota_image_builder/__main__.py

```diff
@@ -0,0 +1,165 @@
+# Copyright 2022 TIER IV, INC. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""External cache source builder helper.
+
+Please refere to https://github.com/tier4/ota-metadata for
+OTA image layout specification.
+This tool is implemented based on ota-metadata@a711013.
+
+Example OTA image layout:
+.
+├── data
+│   ├── <...files with full paths...>
+│   └── ...
+├── data.zst
+│   ├── <... compressed OTA file ...>
+│   └── <... naming: <file_sha256>.zst ...>
+├── certificate.pem
+├── dirs.txt
+├── metadata.jwt
+├── persistents.txt
+├── regulars.txt
+└── symlinks.txt
+
+Check README.md for the offline OTA image layout specification.
+Please refer to OTA cache design doc for more details.
+"""
+
+
+import argparse
+import logging
+import sys
+import tempfile
+from pathlib import Path
+
+from .configs import cfg
+from .builder import build
+from .manifest import ImageMetadata
+
+logger = logging.getLogger(__name__)
+
+
+def main(args):
+    # ------ parse input image options ------ #
+    image_metas = []
+    image_files = {}
+
+    for raw_pair in args.image:
+        _parsed = str(raw_pair).split(":")
+        if len(_parsed) >= 2:
+            _ecu_id, _image_fpath, _image_version, *_ = *_parsed, ""
+            image_metas.append(
+                ImageMetadata(
+                    ecu_id=_ecu_id,
+                    image_version=_image_version,
+                )
+            )
+            image_files[_ecu_id] = _image_fpath
+        else:
+            logger.warning(f"ignore illegal image pair: {raw_pair}")
+
+    if not image_metas:
+        print("ERR: at least one valid image should be given, abort")
+        sys.exit(-1)
```

Comment:
positive value should be used in general. this code is referenced as shell return code.

Reply:
Updated in [95a3341](https://github.com/tier4/ota-client/pull/252/commits/95a3341da83a815ebe4eeb6c2240d3d71a7d87c8).
Now this package will exit with `EINVAL` on invalid parameter specified.